engine/src/main/java/io/zeebe/engine/nwe/behavior/BpmnEventSubscriptionBehavior.java

```diff
@@ -380,6 +381,56 @@ public void unsubscribeFromEvents(final BpmnElementContext context) {
         context.getElementInstanceKey(), context.toStepContext());
   }
 
+  public void triggerEventSubProcess(
+      final ExecutableStartEvent startEvent, final BpmnElementContext context) {
+
+    if (stateBehavior.getFlowScopeInstance(context).getInterruptingEventKey() > 0) {
+      // the flow scope is already interrupted - discard this event
+      return;
+    }
+
+    final var flowScopeContext = stateBehavior.getFlowScopeContext(context);
+
+    triggerEvent(
+        flowScopeContext,
+        eventTrigger -> {
+          final var eventSubProcessElementId = startEvent.getEventSubProcess();
+          final var record =
+              getEventRecord(context.getRecordValue(), eventTrigger, BpmnElementType.SUB_PROCESS)
+                  .setElementId(eventSubProcessElementId);
+
+          final long eventElementInstanceKey = keyGenerator.nextKey();
+          if (startEvent.interrupting()) {
+
+            unsubscribeFromEvents(flowScopeContext);
+
+            final var noActiveChildInstances =
+                stateTransitionBehavior.terminateChildInstances(flowScopeContext);
+            if (noActiveChildInstances) {
+              // activate interrupting event sub-process
+              publishActivatingEvent(context, eventElementInstanceKey, record);
+
+            } else {
+              // wait until child instances are terminated
+              deferActivatingEvent(flowScopeContext, eventElementInstanceKey, record);
+            }
+
+            stateBehavior.updateFlowScopeInstance(
+                context,
+                flowScopeInstance -> {
+                  flowScopeInstance.spawnToken();
+                  flowScopeInstance.setInterruptingEventKey(eventElementInstanceKey);
+                });
+
+          } else {
+            // activate non-interrupting event sub-process
+            publishActivatingEvent(context, eventElementInstanceKey, record);
+          }
+
+          return eventElementInstanceKey;
+        });
```

Comment:
The lambda above is very large, can we find a way to make this a little cleaner? Perhaps extract the body of the lambda to a private method or so?